### DISCUSSION TITLE
ONVIF Support for Restream on Frigate

### DISCUSSION THREAD

**Discussion by moogoos** [blakeblackshear/frigate#15025] [Ask A Question] [answered]
I absolutely love frigate and plan to keep it my primary NVR platform. I also have ubiquiti hardware and they recently released protect to support ONVIF. I currently run a bunch of hikvision cameras and ONVIF works well so now have streams to both Ubiquiti and Frigate go2rtc from the camera. I want to remove the load off the camera for the ONVIF feed and ideally try and tap into Frigate's go2rtc stream. Is this possible?

**Comment by hawkeye217** [comment]
go2rtc supports ONVIF discovery [as documented here](https://github.com/AlexxIT/go2rtc?tab=readme-ov-file#source-onvif) but I'm not sure that's what you're looking for. Can you explain further?

**Reply by moogoos** [reply]
hello,

right now protect allows to connect to a camera's ONVIF specifying an ip and credentials:

![Screenshot 2024-11-16 at 2 17 18 PM](https://github.com/user-attachments/assets/9ee3be67-6ca2-4f8d-ba44-cb7e5b21c79b)

Instead of making another connection to the camera as now:

- protect connects to camera via ONVIF
- frigate connects to camera via RTSP

I was hoping I could simply connect to Frigate via ONVIF instead but I don't think the restreams support an ONVIF connection.

**Reply by hawkeye217** [reply]
All ONVIF is doing here is providing an easy way to discover a camera's RTSP feed address. If you already know the RTSP address for your go2rtc streams (which we do), then there is no benefit to using ONVIF.

**Reply by moogoos** [reply]
yeah but Protect's third party supports only ONVIF

**Reply by NickM-27** [reply]
many users use unify cameras via RTSP. Otherwise, you can just connect go2rtc to the camera using onvif and then use the restream the same as any other.

**Reply by hawkeye217** [reply, marked as answer]
Right. Frigate doesn't have an ONVIF server and as far as I know, go2rtc doesn't support this - but you can ask on [their github](https://github.com/AlexxIT/go2rtc).

**Reply by moogoos** [reply]
ok thank you @hawkeye217.

**Reply by distante** [reply]
@moogoos did you asked there? Have you found a solution?

**Comment by Buster14** [comment]
Hi, i noticed it's now possible already in go2rtc to act as onvif server

https://go2rtc.org/internal/onvif/

So is it doable already? now that protect 7.1 supports motion from cameras it would be awesome if we can make frigate act as third party "AI port"
